### PR TITLE
Add corrupt docker image check for e2e reboot test

### PR DESCRIPTION
**Pull request by ajitak**
At times reboot results in corrupt docker image. Adding a test for docker corrupt image while waiting for pods to be ready will make this test less flaky. https://github.com/kubernetes/kubernetes/issues/47219

### CODE REVIEW

File: test/e2e/reboot.go [hunk 1]
```diff
@@ -281,13 +281,40 @@ func rebootNode(c clientset.Interface, provider, name, rebootCmd string) bool {
 	if !framework.CheckPodsRunningReadyOrSucceeded(c, ns, podNames, rebootPodReadyAgainTimeout) {
 		newPods := ps.List()
 		printStatusAndLogsForNotReadyPods(c, ns, podNames, newPods)
-		return false
+		if isDockerImageCorrupt(c, name) {
+			framework.Logf("Docker image corruption detected for ndoe %s, ignoring pod not ready error", name)
```
Comment (xiangpengzhao) [inline test/e2e/reboot.go]: s/ndoe/node

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: test/e2e/reboot.go (hunk 1 above)
Comment (ajitak) [inline test/e2e/reboot.go]: Updated.

File: test/e2e/reboot.go [hunk 2]
```diff
@@ -281,13 +281,40 @@ func rebootNode(c clientset.Interface, provider, name, rebootCmd string) bool {
...
+	}.AsSelector().String()
+	options := metav1.ListOptions{FieldSelector: selector}
+	events, err := c.Core().Events(namespace).List(options)
+
+	if err != nil || events == nil {
+		return false
+	}
+
+	for _, item := range events.Items {
+		if item.Reason == "CorruptDockerImage" {
```
Comment (spxtr) [inline test/e2e/reboot.go]: Where is this string from? A search for it in this repo yields nothing.

**Review by spxtr — COMMENTED** [review]
I'm a little confused. Sometimes a reboot causes docker image corruption... isn't that a bad thing? Shouldn't we fix that rather than paper over the issue in e2e tests?

File: test/e2e/reboot.go (hunk 2 above)
Comment (ajitak) [inline test/e2e/reboot.go]: This string comes from node problem detector https://github.com/kubernetes/node-problem-detector/pull/117

**Comment by ajitak** [comment]
@dchen1107 What is the plan for addressing the root cause?

**Comment by Random-Liu** [comment]
> I'm a little confused. Sometimes a reboot causes docker image corruption... isn't that a bad thing? Shouldn't we fix that rather than paper over the issue in e2e tests?

In GKE, auto-repair should remedy properly e.g. clean up the docker image once that happens. That's also the original plan.
For open source, since it's a docker problem, and we don't have remedy system, we can't do anything except waiting for docker to fix this and ignore the problem in our test for now.

**Pull request closed without merging**